### DISCUSSION TITLE
如何使用 Layout 佈局中的插槽？

### DISCUSSION THREAD

**Discussion by EvanHsieh0415** [vuepress-theme-hope/vuepress-theme-hope#4459] [Q&A] [answered]
如題，我想在 `tocBefore` 中插入透過自訂 frontmatter 頁面資訊，該如何實現？

**Comment by Mister-Hope** [comment]
There is already a demo here https://theme-hope.vuejs.press/guide/advanced/replace.html#using-slots

tocBefore should be similar

**Reply by EvanHsieh0415** [reply]
我想要的是繼承現有的Layout而做修改，不是從0開始自訂（

**Reply by Mister-Hope** [reply]
The demos behavior is what you asked,   It overrides the home page layout  and add a comment service using the bottom slot on the original home page layout.

Things should be the same  If you want to insert content into toc before slot in the normal page layout.

**Reply by EvanHsieh0415** [reply]
![圖片](https://github.com/user-attachments/assets/5df8cd37-ae5f-42cd-8104-f883f67837fc)

他並沒有繼承原本的NormalPages，可能是我哪裡理解錯誤了

**Reply by Mister-Hope** [reply, marked as answer]
The `NormalPage` is a component which you should override with alias, it's not a layout. Overriding components give you ability to apply things to whole site.

If you only want to insert contents in certain pages with a new layout, extends the "Layout.js" and insert with `tocBefore` slot.
